src/Target.h

```diff
@@ -106,7 +106,7 @@ struct Target {
         FeatureEnd = halide_target_feature_end
     };
     Target() : os(OSUnknown), arch(ArchUnknown), bits(0) {}
-    Target(OS o, Arch a, int b, std::vector<Feature> initial_features = std::vector<Feature>())
+    Target(OS o, Arch a, int b, const std::vector<Feature> &initial_features = std::vector<Feature>())
```

Comment:
I have a feeling that const ref args can't take rvals as their defaults on some older compilers. We had to add Scope::empty_scope to work around.

Reply:
> on some older compilers

Any idea which, and if we still need to support them? (In hindsight, should have added comments about versions to e.g. the empty_scope fixes)

Reply:
Any update on this? I went ahead and tested a simple function that uses the same construction on Godbolt

```
#include <vector>

int test(int a, const std::vector<int>& b = std::vector<int>()) {
    for (const auto& x : b) {
        a += x;
    }
    return a;
}
```

This is accepted by the following compilers:

[g++ >= 4.7.1](https://godbolt.org/z/w-pJbV) (earlier versions reject `-std=c++11`)
[clang >= 3.4.1](https://godbolt.org/z/OCwJ1_) (earlier versions fail to find the `vector` header... an error on Goldbot's part?)
[msvc >= 19.14](https://godbolt.org/z/JpormX) (no earlier versions for me to test)
[icc >= 13.0.1](https://godbolt.org/z/YneUDC) (no earlier versions for me to test)

It also looks like it works with early versions of ellcc (which I think is based on clang?), djggp (which I think is based on gcc?), and power64(le) AT12.0.

Reply:
At this point I think this is a safe construction on all the compilers we're actually using.